wordpress-mobile/WordPress-Android
# Fix: Logout after QR Code Flow deeplink

This PR fixes an issue with a logout immediately following a QR Code Auth flow DeepLink action. See this [conversation](p1658414460654459-slack-C0180B5PRJ4) for more details.

As part of the qr code auth deeplink flow, the backstack is recreated in `ActivityLauncher` using the `TaskStackBuilder`. The original code had built the backstack as follows: WPMainActivity -> MeActivity -> QRCOdeAuthActivity. This worked fine and dandy unless the user selects "log out" immediately following a success/cancel out of the qr code flow. The problem is that `WPMainActivity` is not available to observe the `onSiteRemoved` EventBus event. The handling of that event is what triggers the completion of the log out process; which includes clearing the stack and navigating the user back to the Log in view.

Side note: The same behavior exists for deeplinking off of the zendesk notification (help/support). I did not adjust that process with this PR, as I don't want to make a change in which I have zero context.

To fix this issue, I have eliminated the `MeActivity` from the backstack when deeplinking from the QR Code auth flow. Going forward when the user completes or cancel the QR code login flow after deeplinking, they will be returned to the My Site view instead. I am open to exploring other options.

**Steps to recreate the original issue**
- Install/download the latest JP beta
- Log in to the JPAndroid app and then background the app
- Fire up a QR code reader of the camera app if it has QR code reading capability
- Open a incognito browser and navigate to [here](https://wordpress.com/log-in/qr)
- Scan the QR code to log into WPCOM via QR code.
- Follow flow to the end (actually log in)
- At the end of that process, note that you land on the Me screen in the app.
- Tap “Log out of Jetpack” to logout of the app
- Notice that the app still stays on the Me screen and looks logged in. 
- Tapping logout again will bring you back to the Log in view

**To test:**
**Pre-reqs**
- Uninstall any Jetpack or WP apps from your device
- Install the APK attached to this build

**Links Set up**
- On your device, navigate to settings -> manage Apps -> Jetpack Pre-Alpha
- Scroll down to "Open by Default" and tap
- Tap on the "i" icon to open the available verified links
- Select all the links and then navigate back to the app

**Testing steps***
- Log in to the JPAndroid app and then background the app
- Fire up a QR code reader of the camera app if it has QR code reading capability
- Open a incognito web browser and navigate to [here](https://wordpress.com/log-in/qr)
- Scan the QR code to log into WPCOM via QR code.
- Follow the login flow to the end (actually log in - or you could click cancel)
- ✅ Verify that you land on the My Site screen in the app.
- Navigate back to Me
- Tap “Log out of Jetpack” to logout of the app
- ✅ Verify that you are logged out
- Log out of the web WP.com session

**Regression Test** (perform after pre-req/link set up)
- Log in to the JPAndroid app
- Open a incognito browser and navigate to [here](https://wordpress.com/log-in/qr). 
- Navigate to Me -> Scan Login Code
- Scan the QR code (using the scanner launched from the app) to log into WPCOM 
- Follow the login flow to the end (actually log in - or you could click cancel)
- ✅ Verify that you land on the Me view in the app.
- Tap “Log out of Jetpack” to logout of the app
- ✅ Verify that you are logged out
 
## Regression Notes
1. Potential unintended areas of impact
The deeplink or non-deeplink flows do not work as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tested the flow

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
